src/internal/packager/compose.go

```diff
@@ -56,11 +56,22 @@ func getParentComponent(childComponent types.ZarfComponent, everGrowingComposePa
 		parentComponentName = childComponent.Name
 	}
 
+	targetArch := config.GetArch()
 	// Find the parent component from the imported package that matches our arch
 	for _, importedComponent := range importedPackage.Components {
 		if importedComponent.Name == parentComponentName {
-			parentComponent = importedComponent
-			break
+			filterArch := importedComponent.Only.Cluster.Architecture
+
+			// Override the filter if it is set by the child component
+			if childComponent.Only.Cluster.Architecture != "" {
+				filterArch = childComponent.Only.Cluster.Architecture
+			}
+
+			// Only add this component if it is valid for the target architecture.
+			if filterArch == "" || filterArch == targetArch {
+				parentComponent = importedComponent
+				break
+			}
 		}
 	}
 
```

Comment:
Isn't there a chance that you could reach here without parentComponent being set?  Not sure what go would do here if you try to access parentComponent.Import.Path on a non-match.

Reply:
So I played around with this for a while.  There was an existing issue where an invalid component name could be specified, added an error handler for that.  As far as if you specify the wrong arch, it actually won't make it to this code at all, it would fail at the `config.isCompatibleComponent()` check in the config package instead.  That wouldn't be easy to add more logic around, but we do a couple things to help.  First, we specify in the JSON Schema what is valid and not as far as arch values, second, on package create the component would be removed and a log message in debug mode explaining why.  

## JSON Schema validation error
<img width="717" alt="Screen Shot 2022-07-11 at 2 41 53 PM" src="https://user-images.githubusercontent.com/882485/178345354-62d9be93-1ab3-44b6-bbfb-cb3a5efef3a4.png">

## JSON Schema showing only valid options in drop-down
<img width="707" alt="Screen Shot 2022-07-11 at 2 42 01 PM" src="https://user-images.githubusercontent.com/882485/178345365-3143d774-72ab-4a6d-9305-743e06f03a2f.png">

## Debug warning of component removal
<img width="784" alt="Screen Shot 2022-07-11 at 2 40 08 PM" src="https://user-images.githubusercontent.com/882485/178345462-e58b1dca-70fc-4e9a-8571-d06109577ae9.png">